#! /bin/sh /usr/share/dpatch/dpatch-run
## opt-27-x_subtitles-ttxtsubs-volumebar-fix.dpatch by Thomas Günther <tom@toms-cafe.de>:
## 
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes bug in opt-27-x_subtitles-ttxtsubs which causes crash with
## DP: text2skin + replaybar + volumebar
## DP: (http://www.vdrportal.de/board/thread.php?postid=351009#post351009)

@DPATCH@
diff -Nur vdr-1.4.1/vdr.c vdr-1.4.1.new/vdr.c
--- vdr-1.4.1/vdr.c	2006-06-27 18:34:24.000000000 +0200
+++ vdr-1.4.1.new/vdr.c	2006-06-27 18:35:09.000000000 +0200
@@ -975,7 +975,7 @@
                   }
                else
                   cDevice::PrimaryDevice()->SetVolume(NORMALKEY(key) == kVolDn ? -VOLUMEDELTA : VOLUMEDELTA);
-               if (!Menu /*&& !cOsd::IsOpen()*/)
+               if (!Menu && !cOsd::IsOpen())
                   Menu = cDisplayVolume::Create();
                cDisplayVolume::Process(key);
                key = kNone; // nobody else needs to see these keys
